src/verify.c

```diff
@@ -993,20 +1089,42 @@ enum swupd_code verify_main(void)
 		}
 	}
 
-	set_subscription_versions(official_manifest, NULL, &subs);
-	official_manifest->submanifests = recurse_manifest(official_manifest, subs, NULL, false, NULL);
-	if (!official_manifest->submanifests) {
+	set_subscription_versions(official_manifest, NULL, &all_subs);
+	set_subscription_versions(official_manifest, NULL, &bundles_subs);
+	all_submanifests = recurse_manifest(official_manifest, all_subs, NULL, false, NULL);
+	if (!all_submanifests) {
 		error("Cannot load MoM sub-manifests\n");
 		ret = SWUPD_RECURSE_MANIFEST;
 		goto clean_and_exit;
 	}
+	official_manifest->submanifests = all_submanifests;
+	if (cmdline_option_bundles) {
+		bundles_submanifests = recurse_manifest(official_manifest, bundles_subs, NULL, false, NULL);
+		if (!bundles_submanifests) {
+			error("Cannot load MoM sub-manifests\n");
+			ret = SWUPD_RECURSE_MANIFEST;
+			goto clean_and_exit;
+		}
+		official_manifest->submanifests = bundles_submanifests;
+	}
 	progress_complete_step();
 	timelist_timer_stop(globals.global_times); // closing: Load manifests
 
 	timelist_timer_start(globals.global_times, "Consolidate files from bundles");
 	progress_set_step(4, "consolidate_files");
-	official_manifest->files = files_from_bundles(official_manifest->submanifests);
-	official_manifest->files = consolidate_files(official_manifest->files);
+	all_files = files_from_bundles(all_submanifests);
+	all_files = consolidate_files(all_files);
+	official_manifest->files = all_files;
+	if (cmdline_option_bundles) {
+		bundles_files = files_from_bundles(bundles_submanifests);
+		bundles_files = consolidate_files(bundles_files);
+		/* we need to make sure the files from the selected bundles
+		 * are compared against the full list of consolidated files
+		 * from all bundles so we don't end up deleting a file that
+		 * is needed by another bundle */
+		sanitized_files = sanitize_files(bundles_files, all_files);
```

Comment:
You don't use the variable sanitized_files anywhere else so you don't need to keep it.

Reply:
And the only reason we keep the bundle_files is to free that later. We can change sanitize files to be a filter that frees the items we don't need anymore. So we don't need to keep the pointer to free it later. And probably we can free all_files here too.

Reply:
I would try to simplify this a bit like:

    official_manifest->files = ...
    if (cmdline_option_bundles) {
        bundles_files = consolidated_files_from_bundles(bundles_submanifests);
        //get needed items from all_files and free the ones we don't need. Don't change the bundle_files list.
        official_manifest->files = filter_sanitize_files(official_manifest->files, bundle_files);
        free_list(bundle_files);
    }

Reply:
Done